### PR TITLE
Don't generate birth number of '000' for Swedish personal identity number

### LINKED ISSUE #305: Birth numbers of '000' is not valid for Swedish personal identity number
### Summary

Birth numbers of '000' is not valid for Swedish personal identity numbers. This causes problems in our unit tests. 

The birth number is the first 3 numbers of the last 4, marked with X here 19900101-XXXN

This is probably the most used package when you work with Swedish personal identity numbers and this does not validate the numbers produced: https://github.com/personnummer/php
It specifically fails on this regex: https://github.com/personnummer/php/blob/master/src/Personnummer.php#L124

### Versions

|                | Version |
|:---------------|:--------|
| PHP            | 8.0.3   |
| `fakerphp/faker`  | v1.13.0  |

### Self-enclosed code snippet for reproduction

```php
$this->faker->personalIdentityNumber();
```

### Expected output

This should produce a valid personal identity number with birth number > 000.

### Actual output

Once per 1000 runs on average, this will result in a personal identity number with 000 as birth number, for example: `19640825-0006`. This is not valid.

### PR DESCRIPTION
**Pull request by pelmered**
### What is the reason for this PR?

- [ ] A new feature
- [x] Fixed an issue (resolve #305)

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [x] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Summary of changes

Swedish personal identity number with a birth number of 000 is not valid. The birth number is the first 3 numbers of the last 4, marked with X here 19900101-XXX0

Note: There are no tests for this. I'm not sure how to test this in a good way as this only happen one time in 1000 on average. Any suggestions for this, or can we skip the test?

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

**Comment by pimjansen** [comment]
Missing tests for this

File: src/Faker/Provider/sv_SE/Person.php [hunk 1]
```diff
@@ -132,17 +132,30 @@ public function personalIdentityNumber(\DateTime $birthdate = null, $gender = nu
...
+     */
+    protected function getBirthNumber($gender = null)
+    {
+        do {
+            if ($gender && $gender == static::GENDER_MALE) {
+                $randomDigits = (string) static::numerify('##') . static::randomElement([1, 3, 5, 7, 9]);
+            } elseif ($gender && $gender == static::GENDER_FEMALE) {
+                $randomDigits = (string) static::numerify('##') . static::randomElement([0, 2, 4, 6, 8]);
+            } else {
+                $randomDigits = (string) static::numerify('###');
```
Comment (powellblyth) [inline src/Faker/Provider/sv_SE/Person.php:155]: Could tidy this up with an early return `if (!$gender` to make it a little more readable

Comment (pimjansen) [reply src/Faker/Provider/sv_SE/Person.php:155]: Correct, unwanted and unneeded else statement.

**Comment by bramceulemans** [comment]
@pelmered this looks fine once you add the early return mentioned in @powellblyth's reply.

**Comment by pelmered** [comment]
Now with early returns as requested. 
However, I think the code is considerable less readable now than it was before with the else statement.

File: src/Faker/Provider/sv_SE/Person.php [hunk 2]
```diff
@@ -132,17 +132,40 @@ public function personalIdentityNumber(\DateTime $birthdate = null, $gender = nu
...
+    }
+
+    /**
+     * @param string $gender Person::GENDER_MALE || Person::GENDER_FEMALE
+     *
+     * @return string of three digits
+     */
+    protected function getBirthNumber($gender = null)
+    {
         if ($gender && $gender == static::GENDER_MALE) {
```
Comment (pimjansen) [inline src/Faker/Provider/sv_SE/Person.php:136]: Can we make the checks type safe? I agree that we can force a string due to a breaking change but we could force the type with an ===

Please note that there are multiple occurrences in this class

**Review by pimjansen — CHANGES_REQUESTED** [review]

**Comment by pimjansen** [comment]
> @bramceulemans 
> 
> Updated the code now with early returns as requested. 
> 
> However, I think the code is considerable less readable now than it was before with the else statement. 

Still a small change requested. For the rest it looks fine for me

**Review by bramceulemans — CHANGES_REQUESTED** [review]
If statements should be type-safe.

File: src/Faker/Provider/sv_SE/Person.php (hunk 2 above)
Comment (pelmered) [reply src/Faker/Provider/sv_SE/Person.php:136]: Should I fix the whole class or just this method?

Comment (pelmered) [reply src/Faker/Provider/sv_SE/Person.php:136]: It was only in this method. It's fixed now.

**Review by pimjansen — APPROVED** [review]

**Review by bramceulemans — APPROVED** [review]
LGTM! Thanks for the effort of improving this PR.

**Comment by bramceulemans** [comment]
@pimjansen what do you think about the suggestions from php-cs-fixer?

**Comment by pimjansen** [comment]
> @pimjansen what do you think about the suggestions from php-cs-fixer?
> 
> 

Well not really an opinion though. There are pro and cons for both. But we have the fixer so lets follow it.

**Comment by bramceulemans** [comment]
Thanks for the quick fix.

**Comment by pelmered** [comment]
Thanks for the merge!

**Pull request merged**